Sources/ProjectDescription/AggregateTarget.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+/// An aggregate target of a project.
+public struct AggregateTarget: Codable, Equatable {
```

Comment:
Why do we need a new type, can't it just be a `product`?
Wouldn't it be more consistent with what SPM does?

Reply:
See the reasoning above: https://github.com/tuist/tuist/pull/5266#issuecomment-1614934129

Alternatively, we need to change the `Target` itself. We can make bundle id and platform properties optional and provide two different inits or builders – one for aggregate and one for non-aggreagate targets (where we'd require the bundle id and platform, so users can't make invalid targets).

Reply:
Makes sense 👍